poly-commitment/src/kzg.rs

```diff
@@ -401,25 +402,35 @@ impl<
             .full_srs
             .commit_non_hiding(&eval_polynomial(elm, &evals), 1)
             .elems[0]
-            .into_projective();
+            .into_group();
         let numerator_commitment = { poly_commitment - eval_commitment - blinding_commitment };
         // We compute the result of the multiplication of two miller loop,
         // to apply only one final exponentation
-        let to_loop = [
-            (
-                ark_ec::prepare_g1::<Pair>(numerator_commitment),
-                ark_ec::prepare_g2::<Pair>(Pair::G2Affine::prime_subgroup_generator()),
-            ),
-            (
-                // Note that we do a neagtion here, to put everything on the same side
-                (self.quotient).neg().into(),
-                ark_ec::prepare_g2::<Pair>(divisor_commitment),
-            ),
+        let to_loop_left = [
+            ark_ec::pairing::prepare_g1::<Pair>(numerator_commitment),
+            // Note that we do a neagtion here, to put everything on the same side
+            ark_ec::pairing::prepare_g1::<Pair>(self.quotient.into_group().neg()),
+        ];
+        let to_loop_right = [
+            ark_ec::pairing::prepare_g2::<Pair>(Pair::G2Affine::generator()),
+            ark_ec::pairing::prepare_g2::<Pair>(divisor_commitment),
         ];
         // the result here is numerator_commitment * 1 - quotient * divisor_commitment
         // Note that the unwrap cannot fail as the output of a miller loop is non zero
-        let res = Pair::final_exponentiation(&(Pair::miller_loop(&to_loop))).unwrap();
-
-        res == Pair::Fqk::one()
+        let res = Pair::final_exponentiation(Pair::multi_miller_loop(to_loop_left, to_loop_right))
+            .unwrap();
+
+        res.0 == Pair::TargetField::one()
```

Comment:
Checking this.

Reply:
You can use `res.is_zero()`, see
```
impl<P: Pairing> Zero for PairingOutput<P> {
    /// The identity element, or "zero", of the group is the identity element of the multiplicative group of the underlying field, i.e., `P::TargetField::one()`.
    fn zero() -> Self {
        Self(P::TargetField::one())
    }

    fn is_zero(&self) -> bool {
        self.0.is_one()
    }
}
```

It is better to rely on the interface provided by the library than the inner representation (c.f. affine vs projective coordinates where we can reference `x` and `y` in both structures).